Tests/Realm.Tests/Database/RealmDictionaryTests.cs

```diff
@@ -1133,7 +1133,7 @@ public void DictionaryAsRealmQueryable_RaisesNotifications()
             Assert.That(changeSet.InsertedIndices.Length, Is.EqualTo(2));
             Assert.That(changeSet.DeletedIndices.Length, Is.EqualTo(2));
             Assert.That(changeSet.ModifiedIndices.Length, Is.EqualTo(1));
-            Assert.That(oldDogs.ElementAt(changeSet.ModifiedIndices[0]).Name, Is.EqualTo("foo"));
+            Assert.That(oldDogs.ElementAt(changeSet.NewModifiedIndices[0]).Name, Is.EqualTo("foo"));
```

Comment:
Why this change?

Reply:
It was failing on CI. The index of the newly inserted elements now is different than before. Anyways before it was working by chance, given that we were checking the order in the previous collection instead of the new one